cmd/jaeger/internal/exporters/storageexporter/sanitizer/utf8_sanitizer.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package sanitizer
+
+import (
+	"unicode/utf8"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const (
+	invalidOperation = "InvalidOperationName"
+	invalidService   = "InvalidServiceName"
+	invalidTagKey    = "InvalidTagKey"
+	badUTF8Prefix    = "bad_utf8_"
+)
+
+// NewUTF8Sanitizer creates a UTF8 sanitizer.
+func NewUTF8Sanitizer() SanitizeTraces {
+	return sanitizeUF8
+}
+
+// sanitizeUTF8 sanitizes the UTF8 in the spans.
+func sanitizeUF8(traces ptrace.Traces) ptrace.Traces {
+	resourceSpans := traces.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		resourceSpan := resourceSpans.At(i)
+		scopeSpans := resourceSpan.ScopeSpans()
+		for j := 0; j < scopeSpans.Len(); j++ {
+			spans := scopeSpans.At(j).Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				// Sanitize operation name
+				if !utf8.ValidString(span.Name()) {
+					originalName := span.Name()
+					span.SetName(invalidOperation)
+					byteSlice := span.Attributes().PutEmptyBytes(badUTF8Prefix + "operation_name")
+					byteSlice.FromRaw([]byte(originalName))
+				}
+
+				// Sanitize service name attribute
+				attributes := span.Attributes()
+				serviceNameAttr, ok := attributes.Get("service.name")
```

Comment:
1. service name is a resource attribute, not span attribute
2. you need to call sanitizeAttributes not just on span, but on resource and scope
3. if you do (2) you don't need to special case service name at all (it was different in Jaeger where it was a proper field in the Process)